src/sentry/relay/config/metric_extraction.py

```diff
@@ -537,9 +537,11 @@ def _widget_query_stateful_extraction_enabled(widget_query: DashboardWidgetQuery
     this assumes stateful extraction can be used, and returns the enabled state."""
 
     stateful_extraction_version = OnDemandMetricSpecVersioning.get_default_spec_version().version
-    on_demand_entries = widget_query.dashboardwidgetqueryondemand_set.filter(
```

Comment:
Do we have a perf problem here? or just that we get a lot of spans?
Should we report anything to the SDK teams OR the Django project?
IIRC I don't think this is the first time you have encountered this.

Reply:
Perf problem (N+1). It's gets _around_ the usual fix for n+1 which is `prefetch_related` since the filter means the previously fetched data in cache is ignored. Since we're dealing with very small number of objects, it's totally fine to re-use the cache and do this in memory.